integration-tests/library/build.gradle

```diff
@@ -12,6 +12,10 @@ if (libs.versions.config.generateDaggerFactoriesWithAnvil.get().toBoolean()) {
 
   dependencies {
     kapt libs.dagger2.compiler
+
+    // Necessary because this is what dagger uses when it runs to support instantiating annotations at runtime
+    implementation libs.auto.value.annotations
```

Comment:
I'm curious why this is needed now and not before?

Reply:
tl;dr: [Dagger only creates the annotation creator class for the map key when `@MapKey(unwrapValue = false)` is used.](https://github.com/google/dagger/blob/2067f51a476b1f60c3aa94812af0f8f0db661521/java/dagger/internal/codegen/processingstep/MapKeyProcessingStep.java#L68-L69)

As seen in that link, when `unwrapValue = true`, it uses `UnwrappedMapKeyGenerator` instead, which doesn't use `@AutoAnnotation`.

---

Just to clarify:

Without
```kotlin
implementation libs.auto.value.annotations
kapt libs.auto.value.processor
```

we get
```
> Task :integration-tests:library:kaptKotlin
C:\Projects\anvil\integration-tests\library\build\tmp\kapt3\stubs\main\com\squareup\anvil\test\WrappedBindingKey.java:6: error: @AutoAnnotation is a necessary dependency if @MapKey(unwrapValue = false). Add a dependency for the annotation, "com.google.auto.value:auto-value-annotations:<current version>", and the annotation processor, "com.google.auto.value:auto-value:<current version>"
public abstract @interface WrappedBindingKey {
                 ^
```

Reply:
Thanks for sharing the extra context :+1: